superset/datasets/commands/create.py

```diff
@@ -47,6 +48,7 @@ def run(self) -> Model:
             dataset = DatasetDAO.create(self._properties, commit=False)
             # Updates columns and metrics from the dataset
             dataset.fetch_metadata(commit=False)
+            create_table_permissions(dataset)
```

Comment:
aren't these being created on `security_manager.dataset_after_insert` ?

Reply:
Oh you're right, I'm assuming that `create_table_permissions` came before that function but wasn't removed once `dataset_after_insert` made it unnecessary.  I've removed that function entirely now, let me know if we should keep it though.